engine/access/rpc/backend/connection_factory.go

```diff
@@ -200,7 +214,7 @@ func (s *CachedClient) Close() {
 		return
 	}
 	// allow time for any existing requests to finish before closing the connection
-	time.Sleep(s.timeout)
+	time.Sleep(s.timeout + 1*time.Second)
```

Comment:
why do we need this additional second?

Reply:
Not the most elegant solution but I wanted to ensure that there was allotted time for existing requests to finish OR timeout before a connection is closed. It made sense for the delay in closing to be greater than the timeout time vs just being equal